sdk/src/main/java/org/coralibre/android/sdk/internal/bluetooth/BleClient.java

```diff
@@ -149,11 +149,11 @@ public void onScanFailed(int errorCode) {
     private void onDeviceFound(ScanResult scanResult) {
         try {
             long now = System.currentTimeMillis();
-            ENNumber currentInterval = CryptoModule.getCurrentInterval();
+            ENInterval currentInterval = ENIntervalUtil.getCurrentInterval();
             BluetoothDevice bluetoothDevice = scanResult.getDevice();
             final String deviceAddr = bluetoothDevice.getAddress();
 
-            byte rssi = (byte) scanResult.getRssi();
+            byte rssi = new Integer(scanResult.getRssi()).byteValue();
```

Comment:
In general, you should never use the Integer constructor. Additionally, boxing a primitive int to truncate it to a byte is not necessary. Casting the int to byte gives you the same result more efficiently.

```suggestion
            byte rssi = (byte) scanResult.getRssi();
```